### PR TITLE
intel_debug: add suggestions for DMIC and ES8336 platforms

### PR DESCRIPTION
**Pull request by plbossart**
Instead of repeating the same things on GitHub issues, maybe this
documentation will help.

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

**Review by lgirdwood — COMMENTED** [review]
LGTM, I think @marc-hb has been looking at the recent content generation issues and the pip issue may be related.

**Comment by marc-hb** [comment]
https://github.com/thesofproject/sof-docs/runs/7222607705

I submitted a fix in sof for these 2 doxygen errorsplease help review and expedite: https://github.com/thesofproject/sof/pull/5995
```
/home/runner/work/sof-docs/sof-docs/api/dai-drivers-api.rst:4: WARNING: Duplicate ID: "group__sof__dai__drivers_1ga8c720c310f408b2f97ce014562d6a910".
/home/runner/work/sof-docs/sof-docs/api/dai-drivers-api.rst:4: WARNING: Duplicate explicit target name: "group__sof__dai__drivers_1ga8c720c310f408b2f97ce014562d6a910".

/home/runner/work/sof-docs/sof-docs/api/dai-drivers-api.rst:6: WARNING: Duplicate declaration, const struct device* dai::drv
```
(Yes the fix is in sof.git, not in sof-docs.git. It's a doxygen-related issue but it shows only in sphinx/sof-docs)

  ------

The other errors clearly come from this PR:
```
/home/runner/work/sof-docs/sof-docs/getting_started/intel_debug/suggestions.rst:155: WARNING: Error in "code-block" directive:
maximum 1 argument(s) allowed, 10 supplied.
.. code-block::
   [    4.301490] sof-audio-pci-intel-tgl 0000:00:1f.3: DMICs detected in NHLT tables: 2
```

> not checked due to problem on Ubuntu 22.04

Exactly why I created this alternative a long time ago, please try it:

```
pip3 -r scripts/requirements-lax.txt
```

File: getting_started/intel_debug/suggestions.rst [hunk 1]
```diff
@@ -140,3 +140,116 @@ to snd_sof module:
...
+sof-hda-generic-2ch-pdm1.tplg. On older distributions, it will be
+necessary to override the file installed in
+/lib/firmware/intel/sof-tplg/sof-hda-generic-2ch.tplg. On kernels
+5.20+ a kernel parameter will be enough with no need to change and
+override installed topology files, e.g.
+
+.. code-block::
+   options snd-sof-pci tplg_filename=sof-hda-generic-2ch-pdm1.tplg
+
+These PDM1 issues are tracked in GitHub with the label 'DMIC-PDM1'.
```
[the remark this answers is not in the document]
Comment (plbossart) [reply getting_started/intel_debug/suggestions.rst]: done

File: getting_started/intel_debug/suggestions.rst [hunk 2]
```diff
@@ -140,3 +140,116 @@ to snd_sof module:
...
+should only attempted by users with working knowledge of the Linux
+audio subsystem and an understanding that playing with hardware
+settings may damage hardware.
+
+In rare cases, some platforms use the MCLK1 signal instead of
+MCLK0. As of July 2022, there is no turn-key solution for those
+platforms.
+
+These ES8336 issues are tracked in GitHub with the label 'codec
+ES8336'.
```
[the remark this answers is not in the document]
Comment (plbossart) [reply getting_started/intel_debug/suggestions.rst]: done

**Comment by plbossart** [comment]
> pip3 -r scripts/requirements-lax.txt

pip3 -r scripts/requirements-lax.txt

Usage:   
  pip3 <command> [options]

no such option: -r

???

**Comment by intelkevinputnam** [comment]
> > pip3 -r scripts/requirements-lax.txt
> 
> pip3 -r scripts/requirements-lax.txt
> 
> Usage: pip3 [options]
> 
> no such option: -r
> 
> ???

try: pip3 install -r scripts/requirements-lax.txt

**Comment by marc-hb** [comment]
> try: pip3 install -r scripts/requirements-lax.txt

Indeed, sorry for the mistake. I just tried this on Ubuntu 22.04 and it worked 100%

**Review by ujfalusi — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]
@plbossart pip issues resolved now ?

**Comment by plbossart** [comment]
> @plbossart pip issues resolved now ?

yes, no issues with the 'requirements.lax', the command above was missing a --user switch

**Review by marc-hb — COMMENTED** [review]
Seems useful!

File: getting_started/intel_debug/suggestions.rst [hunk 3]
```diff
@@ -140,3 +140,134 @@ to snd_sof module:
...
+
+On devices designed for Windows, the presence of the microphone is
+reported as an NHTL endpoint. The SOF Linux driver will report this
+information with a 'dmesg' log such as
+
+.. code-block::
+
+   [    4.301490] sof-audio-pci-intel-tgl 0000:00:1f.3: DMICs detected in NHLT tables: 2
+
+In some instances the number of DMICs is incorrect. The SOF driver
```
[the remark this answers is not in the document]
Comment (plbossart) [reply getting_started/intel_debug/suggestions.rst:159]: I reworded but I don't think we can make any references to the Windows driver, I don't know how they deal with this.

File: getting_started/intel_debug/suggestions.rst [hunk 4]
```diff
@@ -140,3 +140,134 @@ to snd_sof module:
...
+Support for this platform only stated upstream with the kernel
+5.19-rc1. Any attempts with earlier kernels will require backports and
+experimental patches to be added.  In the case of the 8326, the codec
+vendor submitted a driver to the ALSA/ASoC maintainers, which was not
+merged as of July 2022. In this specific case end-users will be forced
+to compile their own kernel.
+
+The SOF driver implemented an automatic detection of the SSP/I2S port
+used by hardware and the presence of digital microphones based on
+platform firmware/NHTL.
```
[the remark this answers is not in the document]
Comment (plbossart) [reply getting_started/intel_debug/suggestions.rst:235]: yes

File: getting_started/intel_debug/suggestions.rst [hunk 5]
```diff
@@ -140,3 +140,134 @@ to snd_sof module:
...
+
+There are however a number of hardware configurations that cannot be
+detected from platform firmware. To work-around this limitation, the
```
[the remark this answers is not in the document]
Comment (plbossart) [reply getting_started/intel_debug/suggestions.rst:242]: ACPI and others.

**Comment by lgirdwood** [comment]
@deb-intel are you able to rebuild and publish this week ? Thanks !

**Comment by deb-intel** [comment]
@lgirdwood Yes, I can publish by my EOD today.

**Comment by marc-hb** [comment]
@plbossart in general while waiting for https://thesofproject.github.io/latest/index.html you can point people at GitHub in the meantime, for instance this page is rendered and looks very good: https://github.com/thesofproject/sof-docs/blob/master/getting_started/intel_debug/suggestions.rst

**Comment by marc-hb** [comment]
cc: [@MrChromebox](https://github.com/MrChromebox)

**Comment by lgirdwood** [comment]
Thanks @deb-intel

**Pull request merged**